down_sample_re_count.py

```diff
@@ -1,47 +1,61 @@
-import synapseclient, os, sys
+from __future__ import print_function
+import synapseclient
+import os
+import sys
 
-def pull_data(synId):
-    #1. pull file into local directory
-    #2. syn = synapseclient.Synapse()
-    syn = synapseclient.Synapse()
-    syn.login(os.environ['synapseUser'],os.environ['synapsePassword'])
-    print("logged in")
-    entity = syn.get(synId,downloadLocation = './')
+
+def pull_data(synId, syn):
+    # 1. pull file into local directory
+    # 2. syn = synapseclient.Synapse()
+    entity = syn.get(synId, downloadLocation='./')
     print("file downloaded")
-    bar = entity.path
-    return(bar)
-    #get file name from the entity object
-    #return...
+    return entity.path, entity.annotations
+
 
 def run_samtools(fileName):
     print("running samtools")
-    str = ''.join(['samtools view -b -s 0.5 ',fileName,' > ',fileName,'_new.bam'])
+    str = ''.join(['samtools view -b -s 0.5 ', fileName,
+                   ' > ', fileName, '_new.bam'])
     os.system(str)
     print("samtools run")
-    str2 = ''.join([fileName,'_new.bam'])
-    return(str2)
+    str2 = ''.join([fileName, '_new.bam'])
+    return str2
+
 
-def run_htseq(fileName,gtfFileName):
-    str = ''.join(['htseq-count -f bam ',fileName,' ',gtfFileName,' > ',fileName,'_counts.txt'])
+def run_htseq(fileName, gtfFileName):
+    str = ''.join(['htseq-count -f bam ', fileName, ' ', gtfFileName,
+                   ' > ', fileName, '_counts.txt'])
     print("running htseq")
     os.system(str)
     print("htseq run")
-    str2 = ''.join([fileName,'_counts.txt'])
-    return(str2)
+    str2 = ''.join([fileName, '_counts.txt'])
+    return str2
 
-def push_to_synapse(fileName,uploadLocation):
-    syn = synapseclient.Synapse()
-    syn.login(os.environ['synapseUser'],os.environ['synapsePassword'])
-    entity = synapseclient.File(fileName,parentId = uploadLocation)
+
+def push_to_synapse(fileName, uploadLocation, synId, annotations, syn):
```

Comment:
do you set annotations in this function?

Reply:
There's a mistake. for some reason the function is interpreting the `used` and `executed` arguments as annotations.

You're right. I never ended up passing the annotations into the synapseclient.File constructor. I'll fix both problems.

Reply:
great, thx